src/lib/core/sequence.c

```diff
@@ -329,6 +333,30 @@ bt_sequence_get_machine_unchecked (const BtSequence * const self,
   return self->priv->machines[track];
 }
 
+/*
+ * bt_sequence_get_nonnull_length:
+ *
+ * Return the length of the sequence minus any trailing rows that are totally
+ * null/empty. Includes all available patterns, even those beyond the "end"
+ * of the song.
+ * 
+ * Returns 0 if all rows are null or sequence length is zero.
+ */
+static gulong
+bt_sequence_get_nonnull_length (const BtSequence * const self) {
+  if (self->priv->len_patterns == 0)
+    return 0;
+  
+  for (gulong i = self->priv->len_patterns; i-- > 0; ) {
+    for (gulong j = 0; j < self->priv->tracks; j++) {
```

Comment:
So far I did not used C99 style lops, but separate variable decls. I thinksicne were >2020 its fair to use them. Should we add AC_PROG_CC_C99 to configure to be safe?
https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/C-Compiler.html

Reply:
Sorry, I tend to write using this feature automatically without thinking. But I agree, requiring C99 support is totally reasonable, so I've added the macro it to configure.ac as you suggested.

Interestingly, it looks like the newest versions of autoconf automatically enable ISO C 2011 or C99 support as part of AC_PROG_CC:

https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.70/html_node/C-Compiler.html#C-Compiler

https://www.mail-archive.com/bug-autoconf@gnu.org/msg04671.html

The variable still needs to be tested though, after AC_PROG_CC.